server/functional-test/src/test/java/org/zanata/feature/dashboard/DashboardTest.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2013, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+package org.zanata.feature.dashboard
+
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.openqa.selenium.By
+import org.zanata.feature.Trace
+import org.zanata.feature.testharness.TestPlan.DetailedTest
+import org.zanata.feature.testharness.ZanataTestCase
+import org.zanata.page.dashboard.DashboardBasePage
+import org.zanata.util.HasEmailRule
+import org.zanata.util.ZanataRestCaller
+import org.zanata.workflow.LoginWorkFlow
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import org.assertj.core.api.Assertions.assertThat
+import org.zanata.util.ZanataRestCaller.buildSourceResource
+import org.zanata.util.ZanataRestCaller.buildTextFlow
+
+/**
+ * @author Alex Eng [aeng@redhat.com](mailto:aeng@redhat.com)
+ */
+@Category(DetailedTest::class)
+class DashboardTest : ZanataTestCase() {
+    @get:Rule
+    val emailRule = HasEmailRule()
+    private var dashboard: DashboardBasePage? = null
```

Comment:
looks like another case for `lateinit`. Best to avoid `!!` where possible.

Reply:
Done, also backticks and throws